### ISSUE TITLE
TrackJs seems to override window.onerror if already set

### ISSUE THREAD

**Issue by slorber** [TrackJs/trackjs-package#4]
I'm testing TrackJS and Errorception to see which fit best my needs.

Here's my integration:

``` javascript
        <!-- Errorception -->
        <script crossorigin>
        (function(_,e,rr,s){_errs=[s];var c=_.onerror;_.onerror=function(){var a=arguments;_errs.push(a);
        c&&c.apply(this,a)};var b=function(){var c=e.createElement(rr),b=e.getElementsByTagName(rr)[0];
        c.src="//beacon.errorception.com/"+s+".js";c.async=!0;b.parentNode.insertBefore(c,b)};
        _.addEventListener?_.addEventListener("load",b,!1):_.attachEvent("onload",b)})
        (window,document,"script","token");
        </script>

        <!-- TrackJs -->
        <script type="text/javascript">window._trackJs = { token: 'token' };</script>
        <script type="text/javascript" src="//d2zah9y47r7bi2.cloudfront.net/releases/current/tracker.js" crossorigin></script>
```

It seems TrackJS overrides Errorception listener because only TrackJS receive the errors

---

``` javascript
        <!-- TrackJs -->
        <script type="text/javascript">window._trackJs = { token: 'token' };</script>
        <script type="text/javascript" src="//d2zah9y47r7bi2.cloudfront.net/releases/current/tracker.js" crossorigin></script>

        <!-- Errorception -->
        <script crossorigin>
        (function(_,e,rr,s){_errs=[s];var c=_.onerror;_.onerror=function(){var a=arguments;_errs.push(a);
        c&&c.apply(this,a)};var b=function(){var c=e.createElement(rr),b=e.getElementsByTagName(rr)[0];
        c.src="//beacon.errorception.com/"+s+".js";c.async=!0;b.parentNode.insertBefore(c,b)};
        _.addEventListener?_.addEventListener("load",b,!1):_.attachEvent("onload",b)})
        (window,document,"script","token");
        </script>
```

By registering Errorception after TrackJS it works better because Errorception does not override onerror set by Trackjs but simply proxy it.

**Comment by slorber** [comment]
see https://errorception.com/docs/onerror

**Comment by toddhgardner** [comment]
Thanks @slorber, you're right we should proxy it. I'll try to get this in place for our next tracker release.

**Comment by toddhgardner** [comment]
Fixed in 2.5.0

**Comment by abhilashlr** [comment]
Hello @toddhgardner,

When I try to incorporate window.onerror = function(error) {} in my code, error is being received as the window object while the original error is being captured in trackJS. This seems to happen after 2.5.0, tested with 2.4.0 and it seems to work fine.

Looks like this is due to a method watchWindowErrors where there is a **d.apply(a,c)** where a is the window object and **c** is the array containing the window object. Shouldn't it send the window.onerror's default params instead, like this => **d.apply(a, [e, f, g, h, k])**, so that any other window.onerror watches still work with trackJS included?

Would be great if you can provide some insights.

Thanks

**Comment by toddhgardner** [comment]
hi @abhilashlr, thanks for letting us know. Will check this out and let you know shortly.

**Comment by toddhgardner** [comment]
@abhilashlr confirmed. patch coming soon.

**Comment by abhilashlr** [comment]
Thank you @toddhgardner

**Comment by toddhgardner** [comment]
Hi @abhilashlr this has been patched in 2.5.1

**Comment by abhilashlr** [comment]
Thanks @toddhgardner, will test and confirm.
